solution/src/main/kotlin/com/cosmotech/solution/service/SolutionServiceImpl.kt

```diff
@@ -222,39 +241,39 @@ class SolutionServiceImpl(
   }
 
   override fun updateSolution(
-      organizationId: String,
-      solutionId: String,
-      solution: Solution
+    organizationId: String,
+    solutionId: String,
+    solutionUpdateRequest: SolutionUpdateRequest
   ): Solution {
     val existingSolution = getVerifiedSolution(organizationId, solutionId, PERMISSION_WRITE)
 
     // solutionId update is allowed but must be done with care. Maybe limit to minor update?
-    var hasChanged =
+
+    val updatedSolution = Solution(
+      id = solutionId,
+      name = solutionUpdateRequest.name ?: existingSolution.name,
+      organizationId = existingSolution.organizationId,
+      ownerId = existingSolution.ownerId,
+      description = solutionUpdateRequest.description,
```

Comment:
Similar to previous PR, all fields from the update request are optional, you need to provide the current value as default if the request does not have them. Otherwise we'll overwrite and loose the current value.